layers/+lang/erlang/funcs.el

```diff
@@ -0,0 +1,52 @@
+;;; funcs.el --- Erlang Layer functions File for Spacemacs
+;;
+;; Copyright (c) 2012-2020 Sylvain Benner & Contributors
+;;
+;; Author: Carlos F. Clavijo <arkan1313@gmail.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+(defun spacemacs//erlang-backend ()
+  "Returns selected backend."
+  (if erlang-backend
+      erlang-backend
+    (cond
+     ((configuration-layer/layer-used-p 'lsp) 'lsp)
+     (t 'company-erlang))))
+
+(defun spacemacs//erlang-setup-backend ()
+  "Conditionally setup erlang backend."
+  (pcase (spacemacs//erlang-backend)
+    (`lsp (spacemacs//erlang-setup-lsp)))
+  )
+
+(defun spacemacs//erlang-setup-company ()
+  "Conditionally setup company based on backend."
+  (pcase (spacemacs//erlang-backend)
+    ;; Activate lsp company explicitly to activate
+    ;; standard backends as well
+    (`lsp (progn
+            (spacemacs|add-company-backends
+              :backends company-lsp
+              :modes erlang-mode
+              :append-hooks nil
+              :call-hooks t)
+            (setq company-minimum-prefix-length 1)
+            (setq company-idle-delay 0.3))))
+  (add-hook 'erlang-mode-hook 'company-mode))
```

Comment:
This should not be directly called instead use `:append-hooks` tag of spacemacs|add-company-backends which will do the same.